app/controllers/users_controller.rb

```diff
@@ -26,4 +37,13 @@ def show
   def user_params
     params.require(:user).permit!
   end
+
+  def handle_invitation
+    if params[:invitation_token]
+      invitation = Invitation.find_by(token: params[:invitation_token])
+      invitation.inviter.follow(@user)
+      @user.follow(invitation.inviter)
+      invitation.update_column(:token, nil)
```

Comment:
I would move `invitation.update_column(:token, nil)` to model layer. You have `generate_token` there, could as well have `remove_token`.